src/docs/guides/storage-buckets.md

```diff
@@ -5,99 +5,78 @@ description: Persist assets in object storage.
 
-You can reference these directly in your service's environment variables without copy-pasting credentials manually.
+<video src="https://res.cloudinary.com/railway/video/upload/v1763419449/VariableReferenceManual_ld79zb.mp4" controls autoplay loop muted playsinline />
+</Collapse>
 
-<Image src="https://res.cloudinary.com/railway/image/upload/v1764373717/docs/bucket_variables_q6vevd.png"
-alt="Screenshot of bucket variables in a service"
-layout="responsive"
-width={1908} height={1378} quality={80} />
+<Collapse slug="storage-buckets-automatic-credentials-variables" title="Automatically provisioning the variables to your service">
+You can insert all the required authentication variables into your service's variables depending on your S3 client. We have presets for the AWS SDK, Bun's built-in S3 driver, FastAPI, Laravel, and more.
 
-To bulk-add all bucket credentials to a service, head to the "Credentials" tab in your bucket and click "Add to Service". You can choose environment variable naming conventions from popular libraries and frameworks (like AWS SDK, Bun, etc.), or define your own variable names.
+Doing this sets the names for the credentials based on what each library expects as the default environment variable names. Supported libraries (notably the official AWS SDKs) can automatically pick them up so you don't have to provide each variable to the S3 client.
 
 <video src="https://res.cloudinary.com/railway/video/upload/v1763419460/AutoInjectVariables_xborrx.mp4" controls autoplay loop muted playsinline />
+</Collapse>
 
 ## Buckets in Environments
 
 Each environment gets its own separate bucket instance with isolated credentials. When you [duplicate an environment](/guides/environments#create-an-environment) or use [PR environments](/guides/environments#enable-pr-environments), you won't need to worry about accidentally deleting production objects, exposing sensitive data in pull requests, or polluting your production environment with test data.
 
 ## How buckets are billed
 
-Buckets are billed as GB-month, based on the total amount of data stored across all bucket instances in your workspace.
+Buckets are billed at **$0.015** per GB per month (30 days), based on the total amount of data stored across all bucket instances in your workspace, including Environments. All S3 API operations are unlimited and free. Egress is also unlimited and free, whether that's using presigned URLs or via the S3 API.
 
-- $0.015 per GB-month of storage
-- S3 API operations (PUT, GET, DELETE, LIST, etc.) are unlimited and free
-- Egress from buckets is unlimited and free
+<Banner variant="info">Even though *buckets* don't charge for ingress or egress, buckets live on the public network and are subject to egress usage from your other Railway services when they upload files to your buckets. Buckets are currently not available on the private network.</Banner>
```

Comment:
> [buckets] are subject to egress usage from your other Railway services

This sounds to me like you have to pay for bucket egress, which isn't the case. Would be good if it can be clearer that _service egress_ occurs when you upload a file to a bucket.